### ISSUE TITLE
Some confusion in training downstream segmentation models

### ISSUE THREAD

**Issue by xrli-U** [sjtuplayer/anomalydiffusion#41]
Can I understand the settings in the training downstream segmentation network by this way ?
1. There are only 500 images per anomlay class for training. But in the paper, per anomaly class has 1000 images.
2. An epoch only randomly selects 500 images for training, in which the number of normal images and abnormal images is equal.
3. The maximum epoch is 200, but the test data is used to select best_epoch. Is this really allowed when training a model?

Looking forward to your answers. Thank you very much!

**Comment by sjtuplayer** [comment]
> Can I understand the settings in the training downstream segmentation network by this way ?
> 
> 1. There are only 500 images per anomlay class for training. But in the paper, per anomaly class has 1000 images.
> 2. An epoch only randomly selects 500 images for training, in which the number of normal images and abnormal images is equal.
> 3. The maximum epoch is 200, but the test data is used to select best_epoch. Is this really allowed when training a model?
> 
> Looking forward to your answers. Thank you very much!

1. We have abandoned 500 image-pairs with lower quality as illustrated in the reamdme. If you need the other 500 images, you can use the offered pretrained model to generate them. But the 500-version is more stable when training.
2. Yes, the numbers are equal.
3. This is not the most standard way to select the model for a supervised model. But since there is not a validation set in anomaly detection task, it is hard to choose the model we need in a supervised training process. Since most of the previous methods are unsupervised methods, they do not have such a problem. For supervised-based anomaly detection, there may be a better way to select the model in the future. (Maybe one way is to use the generated data as the validation set)

**Comment by gcs1813** [comment]
@xrli-U , even though the models are validated on the test, the results are not reproduceable, @sjtuplayer  is very kind guy, I appreciate him, he always replies and is very cooperative, but there are a lot mysteries in this work. F1- max for anomaly segmentation could be computed with publicly available sklearn functions, but I don't know why authors choose manual process. Based on the reported it is very tough even to achieve same results using their own framework. The authors should clarify these issues so that other pears who are working in similar can do experiments easily.   @sjtuplayer , @Jiangning Zhang, @ran yi
